src/Endpoints/HttpFactoryTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * HttpFactoryTrait
+ *
+ * @since     Jan 2024
+ * @author    Haydar KULEKCI <haydarkulekci@gmail.com>
+ */
+
+namespace Qdrant\Endpoints;
+
+use Http\Discovery\Psr17FactoryDiscovery;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Qdrant\Exception\InvalidArgumentException;
+
+trait HttpFactoryTrait
+{
+    protected ?RequestFactoryInterface $httpFactory = null;
+
+    public function getHttpFactory(): RequestFactoryInterface
+    {
+        if ($this->httpFactory === null) {
+            $this->httpFactory = Psr17FactoryDiscovery::findRequestFactory();
```

Comment:
As we require `psr/http-factory-implementation` then - I think - we can avoid `Http\Discovery\Psr17FactoryDiscovery` requirement.

Reply:
As I mentioned in another comment, let's merge this and try to improve it in another PR before release.